### DISCUSSION TITLE
CSV Import for transactions

### DISCUSSION THREAD

**Discussion by mkitzmann** [GoPlan-Finance/goplan-app#142] [Ideas]
We could implement a csv import for transactions, so data can be added faster.
The application I am currently using for my portfolio tracking "Portfolio Performance" is using these headers for their csv export:

Date | Type | Value | Transaction Currency | Taxes | Shares | ISIN | WKN | Ticker Symbol | Security Name | Note
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --

(WKN is the german asset identification number)

Where `Type` is one of the following:
- Buy
- Sell
- Removal
- Deposit

Since there are a couple of people using this tool, I would suggest using this format for csv import to simplify the transition for "Portfolio Performance" users.

**Comment by sadortun** [comment]
Is `Deposit` for dividends ? If not, where are they accounted for ?

**Reply by mkitzmann** [reply]
Deposit could be used for that, i will look into how it is used.
Also "Portfolio Performance" exports one csv for every account, so the account name is also missing here.

**Reply by mkitzmann** [reply]
Im not sure how Taxes and Fees are handled. They might be part of the 'Buy' and 'Sell' transactions. But that would lead to inconsistencies.

**Reply by sadortun** [reply]
We'll not handle tax and fees until we have then`Accounts` in place. 

We could import the data right away, but it won't be used.

**Comment by mkitzmann** [comment]
So, I had another look at "Portfolio Performance" CSV exports.
It exports one csv for all transactions of an account and one csv for all transaction of a portfolio.
The account transactions have the headers stated above and "Dividend" is handled as a seperate Type.
The portfolio transaction has the following headers:

- Date	
- Type	
- Value	
- Transaction Currency	
- Gross Amount	
- Currency Gross Amount	
- Exchange Rate	
- Fees	
- Taxes	
- Shares	
- ISIN	
- WKN	
- Ticker Symbol	
- Security Name	
- Note

Where there are only `Buy` and `Sell` types and the Fees and Taxes are part of one transaction.

**Reply by sadortun** [reply]
Just upload a sample file and I'll have a look at it :) 

What I had in mind for the import for now is only to import :
- type
- symbol
- qty
- date

Im only implementing it to be able to test other app features, and not have to add 500 transactions by hand :)

**Reply by mkitzmann** [reply]
For a start having a simplified import would be totally fine for me. This was meant more as a mid term feature idea. 
I will send you a sample csv.
